src/twisted/application/internet.py

```diff
@@ -1029,13 +1058,31 @@ def __init__(self, endpoint, factory, retryPolicy=None, clock=None):
             this attribute will not be serialized, and the default value (the
             reactor) will be restored when deserialized.
         @type clock: L{IReactorTime}
+
+        @param prepareConnection: A single argument L{callable} that may return
+            a L{Deferred}. It will be called once with the L{protocol
+            <interfaces.IProtocol>} each time a new connection is made.  It may
+            call methods on the protocol to prepare it for use (e.g.
+            authenticate) or validate it (check its health).
+
+            The C{prepareConnection} callable may raise an exception or return
+            a L{Deferred} which fails to reject the connection.  A rejected
+            connection is not used to resolve an L{Deferred} returned by
```

Comment:
Deferreds are generally said to be "fired" rather than "resolved" (which is the language one uses with JavaScript Promises). This isn't a dealbreaker, use this language if you prefer it, it's just a little inconsistent with the rest of Twisted.